grammars/javascript.cson

```diff
@@ -977,6 +1005,132 @@
         'name': 'variable.parameter.function.js'
       }
     ]
+  'arguments':
+    'patterns': [
+      {
+        'include': '#method_call'
+      }
+      {
+        'include': '#function_call'
+      }
+      {
+        'include': '#numbers'
+      }
+      {
+        'include': '#strings'
+      }
+      {
+        'include': '#object_variable'
+      }
+      {
+        'match': '\\[|\\]'
+        'name': 'meta.brace.square.js'
+      }
+      {
+        'match': ','
+        'name': 'meta.delimiter.object.comma.js'
+      }
+      {
+        'match': '\\{|\\}'
+        'name': 'meta.brace.curly.js'
+      }
+      {
+        'include': '#operators'
+      }
+      {
+        'include': '#comments'
+      }
+      {
+        'include': '#properties'
+      }
+      {
+        'match': '(?x) \\b[a-zA-Z_$]\\w* (?= \\s*, | \\) )'
+        'name': 'variable.other.argument.js'
+      }
+      {
+        'comment': 'e.g. 123arg'
+        'match': '(?x) \\b\\d\\w+ (?= \\s* (?: , | \\) ) )'
+        'name': 'invalid.illegal.js'
+      }
+    ]
+  'method_call':
+    'patterns': [
+      {
+        'comment': '.methodCall(arg1, "arg2", [...])'
+        'begin': '(?x) (\\.) (\\w+) \\s* (\\()'
+        'beginCaptures':
+          '1':
+            'name': 'meta.delimiter.method.period.js'
+          '2':
+            'name': 'entity.name.function.js'
+          '3':
+            'name': 'punctuation.definition.arguments.begin.js'
+        'end': '\\)'
+        'endCaptures':
+          '0':
+            'name': 'punctuation.definition.arguments.end.js'
+        'name': 'meta.method-call.js'
+        'patterns': [
+          {
+            'include': '#arguments'
+          }
+        ]
+      }
+    ]
+  'function_call':
+    'patterns': [
+      {
+        'comment': 'functionCall(arg1, "arg2", [...])'
+        'begin': '(?x) \\b(\\w+) \\s* (\\()'
+        'beginCaptures':
+          '1':
+            'name': 'entity.name.function.js'
+          '2':
+            'name': 'punctuation.definition.arguments.begin.js'
+        'end': '\\)'
+        'endCaptures':
+          '0':
+            'name': 'punctuation.definition.arguments.end.js'
+        'name': 'meta.function-call.js'
+        'patterns': [
+          {
+            'include': '#arguments'
+          }
+        ]
+      }
+    ]
+  'object_variable':
+    'patterns': [
+      {
+        'comment': '`obj` in `obj.prop`, `obj.methodCall()`'
+        'captures':
+          '1':
+            'name': 'variable.other.object.js'
+          '2':
+            'name': 'invalid.illegal.js'
+        'match': '(?x)
```

Comment:
Couldn't this be `\\b([A-Za-z_$]\\w*)|(\\d+\\w+)(?=\\x*\\.)`?
(Also, I think the `\\d` should be `\\d+` instead to capture `123a` and not just `1a`))